sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -119,7 +119,7 @@ abstract class Optimizer(catalogManager: CatalogManager)
         rulesWithoutInferFiltersFromConstraints: _*) :: Nil
     }
 
-    (Batch("Eliminate Distinct", Once, EliminateDistinct) ::
+    val batches = (Batch("Eliminate Distinct", Once, EliminateDistinct) ::
```

Comment:
I get it that we want to keep the diff small so we leave the indentation wrong here, but we can avoid this problem completely.

An empty batch has nearly 0 overhead. The `RuleExecutor` just runs an empty loop and goes to the next batch. The join-reorder optimization can be turned off and we still keep its batch.

Reply:
Keeping the empty batch causes test failures because tests assume that it will be filtered out. It's better to filter it out here.